bitshares-ubuntu-vm/azuredeploy.json

```diff
@@ -0,0 +1,241 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "newStorageAccountName": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Storage Account where the Virtual Machine's disks will be placed."
+      }
+    },
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Username for the Virtual Machine."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "dnsNameForPublicIP": {
+      "type": "string",
+      "metadata": {
+        "description": "Unique DNS Name for the Public IP used to access the Virtual Machine."
+      }
+    },
+    "location": {
+      "type": "string",
+      "allowedValues": [
+        "East US",
+        "West US",
+        "West Europe",
+        "East Asia",
+        "South East Asia"
+      ],
+      "defaultValue": "West US",
+      "metadata": {
+        "description": "Deployment location"
+      }
+    },
+    "vmSize": {
+      "type": "string",
+      "defaultValue": "Standard_A2",
+      "allowedValues": [
+        "Standard_A1",
+        "Standard_A2",
+        "Standard_A3",
+        "Standard_D1_V2",
+        "Standard_D2_V2",
+        "Standard_D3_V2"
+      ],
+      "metadata": {
+        "description": "Size of VM"
+      }
+    }
+  },
+  "variables": {
+    "storageNameLower": "[toLower(parameters('newStorageAccountName'))]",
+    "imagePublisher": "Canonical",
+    "imageOffer": "UbuntuServer",
+    "OSDiskName": "BitSharesOSDisk",
+    "imageSKU": "15.10",
+    "nicName": "BitSharesNic",
+    "addressPrefix": "10.0.0.0/16",
+    "subnetName": "BitSharesSubnet",
+    "subnetPrefix": "10.0.0.0/24",
+    "storageAccountType": "Standard_LRS",
+    "publicIPAddressName": "BitSharesPublicIP",
+    "publicIPAddressType": "Dynamic",
+    "vmStorageAccountContainerName": "vhds",
+    "vmName": "BitSharesVM",
+    "virtualNetworkName": "BitSharesVNET",
+    "vnetID": "[resourceId('Microsoft.Network/virtualNetworks',variables('virtualNetworkName'))]",
+    "subnetRef": "[concat(variables('vnetID'),'/subnets/',variables('subnetName'))]",
+    "apiVersion": "2015-06-15",
+    "environmentSettings": {
+      "Standard_A1": {
+        "coreCount": 1
+      },
+      "Standard_A2": {
+        "coreCount": 2
+      },
+      "Standard_A3": {
+        "coreCount": 4
+      },
+      "Standard_D1_V2": {
+        "coreCount": 1
+      },
+      "Standard_D2_V2": {
+        "coreCount": 2
+      },
+      "Standard_D3_V2": {
+        "coreCount": 4
+      }
+    },
+    "currentEnvironmentSettings": "[variables('environmentSettings')[parameters('vmSize')]]",
+    "coreCount": "[variables('currentEnvironmentSettings').coreCount"
```

Comment:
Interesting use of the object for getting the cores in the script below. One thing I worry is if you introduce more sizes, then you have to update this object. Wouldn't getting number of cores from `nproc` or `cat /proc/cpuinfo` in the shell script be better?